File: web/src/app/core/_models/Article.ts
export class Article {
  id: string;
  title: string;
  description: string = '';
  reference: string;
  availability: string;
  price: number;
  tax:number;
  quantity: number;
  creationDate: Date;
  active: Boolean;
  deleted: Boolean;
  discount: Discount = new Discount();
  images: Media[] = [];
  videos: Media[] = [];
  features: Feature[] = [];
  categories: CategoryTable[] = [];
  priceHistory: PriceHistory[] = [];
}
export class Discount {
  onDiscount: Boolean = false;
  percentage: number = 0;
}
export class Media {
  id:string;
  path: string;
  constructor(path: string) {
    this.path = path;
  }
}
export class Feature {
  id: string;
  name: string = '';
  value: String = '';
}
export class CategoryTable {
  category: string;
  subCategory: String;
}
export class PriceHistory {
  price: Number;
  updateDate: Date;
  constructor(price: Number) {
    this.price = price;
    this.updateDate = new Date();
  }
}

File: service-article-read/src/main/java/com/example/servicearticleread/entites/Article.java
package com.example.servicearticleread.entites;

import coreapi.models.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.persistence.Id;
import java.util.Date;
import java.util.List;

@Data @NoArgsConstructor @AllArgsConstructor
@Document(collection = "article_db")
public class Article {
    @Id
    private String id;
    private String title;
    private String description;
    private String reference;
    private String availability;
    private double price;
    private double tax;
    private int quantity;
    private Date creationDate;
    private Boolean active;
    private Boolean deleted;
    private Discount discount;
    private List<Image> images;
    private List<Video> videos;
    private List<Feature> features;
    private List<Category> categories;
    private List<PriceHistory> priceHistory;


}
